dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -130,6 +134,32 @@ public void run(@Nonnull ConnectorArguments arguments) throws Exception {
   private <T> T runTask(
       @Nonnull TaskRunContext context, @Nonnull TaskSetState.Impl state, Task<T> task)
       throws MetadataDumperUsageException {
+    T t = executeTask(context, state, task);
+
+    numberOfCompletedTasks += 1;
+
+    long averageTimePerTaskInMillisecond =
+        (System.currentTimeMillis() - startTimeInMilliseconds) / numberOfCompletedTasks;
+
+    int percentFinished = numberOfCompletedTasks * 100 / totalNumberOfTasks;
+    String progressMessage = percentFinished + "% Completed";
+    PROGRESS_LOG.info(progressMessage);
+    LOG.info(progressMessage);
+
+    int remainingTasks = totalNumberOfTasks - numberOfCompletedTasks;
+    long estimatedTimeInMinutes = averageTimePerTaskInMillisecond * remainingTasks / (60 * 1000);
+
+    if (estimatedTimeInMinutes > 0 && numberOfCompletedTasks > 10) {
+      String estimationMessage = estimatedTimeInMinutes + " minutes remaining";
+      PROGRESS_LOG.info(estimationMessage);
+      LOG.info(estimationMessage);
+    }
```

Comment:
This is less important, but consider moving all the new code in this method (with the exception of the line `numberOfCompletedTasks += 1;`) to a new method, e.g. `logProgress`.